src/main/java/com/logentries/logger/AndroidLogger.java

```diff
@@ -0,0 +1,41 @@
+package com.logentries.logger;
+
+import android.content.Context;
+
+import java.io.IOException;
+
+public class AndroidLogger {
+
+    private static AndroidLogger instance;
+
+    private AsyncLoggingWorker loggingWorker;
+
+    private AndroidLogger(Context context, boolean useHttpPost, boolean useSsl, boolean isUsingDataHub, String dataHubAddr, int dataHubPort,

```

Comment:
Probably could do with separate constructors here as this one is pretty overkill